packages/expo-router/src/link/zoom/zoom-transition-context.tsx

```diff
@@ -10,3 +20,40 @@ export type ZoomTransitionSourceContextValueType =
 
 export const ZoomTransitionSourceContext =
   createContext<ZoomTransitionSourceContextValueType>(undefined);
+
+export const ZoomTransitionTargetContext = createContext<{ identifier: string | null }>({
+  identifier: null,
+});
+
+export function ZoomTransitionTargetContextProvider({
+  route,
+  children,
+}: PropsWithChildren<{ route: unknown }>) {
+  const isPreview = useIsPreview();
+  if (
+    isZoomTransitionEnabled() &&
+    !isPreview &&
+    route &&
```

Comment:
unrelated to this PR / `useScreens`: I'm pretty sure we have a type for this from react-navigation, but noticed that `useScreens` is also bailing with `any` props. Might be time to revisit this to clarify what kind of `route` type this is, since it isn't a `RouteNode`, and instead the `react-navigation` internal one, but we don't assign it consistently?